modules/ingest-common/src/test/java/org/elasticsearch/ingest/common/IngestRestartIT.java

```diff
@@ -177,6 +186,98 @@ public void testPipelineWithScriptProcessorThatHasStoredScript() throws Exceptio
     }
 
+    public void testDefaultPipelineBeforeCSRecovered() throws Exception {
+        internalCluster().startNode();
+
+        BytesReference pipeline = new BytesArray("{\n" +
+            "  \"processors\" : [\n" +
+            "    {\n" +
+            "      \"remove\": {\n" +
+            "        \"field\": \"_type\"\n" +
+            "      }\n" +
+            "    }" +
+            "  ]\n" +
+            "}");
+        client().admin().cluster().preparePutPipeline("test_pipeline", pipeline, XContentType.JSON).get();
+        client().admin().indices().preparePutTemplate("pipeline_template")
+            .setPatterns(Collections.singletonList("*"))
+            .setSettings(
+                "{\n" +
+                "  \"index\" : {\n" +
+                "     \"default_pipeline\" : \"test_pipeline\"" +
+                "   }\n" +
+                "}\n", XContentType.JSON).get();
+
+
+        internalCluster().fullRestart(new InternalTestCluster.RestartCallback() {
+            @Override
+            public Settings onNodeStopped(String nodeName) {
+                return Settings.builder().put(GatewayService.RECOVER_AFTER_NODES_SETTING.getKey(), "2").build();
+            }
+
+            @Override
+            public boolean validateClusterForming() {
+                return false;
+            }
+        });
+
+        CountDownLatch latch = new CountDownLatch(1);
+
+        // this one should fail
+        assertThat(expectThrows(ClusterBlockException.class, () -> client().prepareIndex("index", "foo", "fails")
+            .setSource("x", 1)
+            .setTimeout(TimeValue.timeValueMillis(100)) // 100ms, to fail quickly
+            .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+            .get()).getMessage(), equalTo("blocked by: [SERVICE_UNAVAILABLE/1/state not recovered / initialized];"));
+
+        // but this one should pass since it has a longer timeout
+        client().prepareIndex("index", "foo", "passes1")
+            .setSource("x", 2)
+            .setTimeout(TimeValue.timeValueSeconds(60)) // wait for second node to start
+            .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+            .execute(new ActionListener<IndexResponse>() {
+                @Override
+                public void onResponse(IndexResponse indexResponse) {
+                    assertThat(indexResponse.status(), equalTo(RestStatus.CREATED));
+                    assertThat(indexResponse.getResult(), equalTo(DocWriteResponse.Result.CREATED));
+                    latch.countDown();
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    fail("Should not have failed with exception: " + e.getMessage());
+                }
+            });
+
+        // so the cluster state can be recovered
+        internalCluster()
+            .startNode(Settings.builder().put(GatewayService.RECOVER_AFTER_NODES_SETTING.getKey(), "1"));
```

Comment:
I think this can just be `.startNode()` - the other node is the master, so this node doesn't do any state recovery.